src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-form/nfs-form.component.ts

```diff
@@ -179,15 +231,44 @@ export class NfsFormComponent extends CdForm implements OnInit {
     const subvol = this.nfsForm.getValue('subvolume');
     if (subvol === '') {
       const subvolGroup = this.nfsForm.getValue('subvolume_group');
-      this.fsPath = this.getFsPath(this.allsubvolgrps, subvolGroup);
+      if (!_.isEmpty(this.selectedSubvolGroup)) {
+        const fs_name = this.nfsForm.getValue('fsal').fs_name;
+        if (this.selectedSubvolGroup === '_nogroup') {
+          this.fsPath = '/volumes/_nogroup';
+        } else {
+          this.getSubVolGrpInfo(fs_name, this.selectedSubvolGroup);
+        }
+      } else {
+        if (this.allsubvolgrps.length === 0) {
+          this.fsPath = '/';
+        } else {
+          if (subvolGroup === '_nogroup') {
+            this.fsPath = '/volumes/_nogroup';
+          } else {
+            this.fsPath = this.getFsPath(this.allsubvolgrps, subvolGroup);
+          }
+        }
+        this.nfsForm.patchValue({
+          path: this.fsPath
+        });
+        this.pathChangeHandler();
+      }
     } else {
-      this.fsPath = this.getFsPath(this.allsubvols, subvol);
+      if (!_.isEmpty(this.selectedSubvol)) {
+        const fs_name = this.nfsForm.getValue('fsal').fs_name;
+        this.getSubVolInfo(fs_name, this.selectedSubvolGroup, this.selectedSubvol);
+      } else {
+        if (this.allsubvols.length === 0) {
+          this.fsPath = '/';
+        } else {
+          this.fsPath = this.getFsPath(this.allsubvols, subvol);
+        }
+        this.nfsForm.patchValue({
+          path: this.fsPath
+        });
+        this.pathChangeHandler();
+      }
```

Comment:
i don't understand the whole function here.. what is it that we are trying to achieve here? How can we simplify it? this is really unmaintainable if we don't understand what's going on. and the fact that this is not unit-tested at all is not really good. can you explain what's happening here?